VirtualReality/Program.cs

```diff
@@ -1,29 +1,155 @@
+            {
+                Console.WriteLine(keyValuePair.ToString());
+            }
+
+            // get user input for which session to connect to
+            Console.WriteLine("Which client should be connected to?");
+            string userInput = Console.ReadLine();
+
+            if (CreateTunnel(userInput))
+            {
+                Console.WriteLine("Succes connected to " + userInput);
+            }
+            else
+            {
+                Console.WriteLine("couldn't connect to that client");
+            }
+
+        }
+
+
+
+        /// <summary>CreateTunnel does <c>Creating a network tunnel</c> returns <returns>A Boolean</returns> sends the correct json and then checks connection status based on that it returns a boolean</summary>
+        public Boolean CreateTunnel(String sessionID)
+        {
+            if (userSessionsMap.ContainsKey(sessionID))
+            {
+                Console.WriteLine("Creating a tunnel");
+                // create a tunnel
+                JObject tunnelCreateJson = new JObject { { "id", "tunnel/create" } };
+
+                JObject dataJson = new JObject { { "session", userSessionsMap.GetValueOrDefault(sessionID) } };
+                // place to set the key 
+                string sessionKey = "";
+                dataJson.Add("key", sessionKey);
+
+                tunnelCreateJson.Add("data", dataJson);
+
+
+                SendToTcp(networkStream, tunnelCreateJson.ToString());
+                string tunnelCreationResponse = "";
 
-            NetworkStream networkStream = client.GetStream();           
-            networkStream.Write(BitConverter.GetBytes(length));
-            networkStream.Write(byteData);
 
-            
 
+                ReceiveFromTcp(networkStream, out tunnelCreationResponse);
+
+
+                dynamic responseDeserializeObject = JsonConvert.DeserializeObject(tunnelCreationResponse);
+                string response = responseDeserializeObject["data"]["status"].ToString();
+
+                if (response != "oke")
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+
+
+        /// <summary>SendToTcp does <c>Sending a String over Tcp</c> using ASCII encoding</summary>
+        public void SendToTcp(NetworkStream networkStream, string data)
+        {
+            byte[] dataBytes = System.Text.Encoding.ASCII.GetBytes(data);
+            int dataLength = dataBytes.Length;
+
+            networkStream.Write(BitConverter.GetBytes(dataLength));
+            networkStream.Write(dataBytes);
+        }
+
+        /// <summary>GetRunningSessions does <c>Getting a all running sessions from the server</c> returns <returns>A Dictionary<string, string> containing all users as key and a value of all data</returns> sends data using SendDataToTCP and then Receive it using ReceiveFromTcp</summary>
```

Comment:
```suggestion
        /// <summary>GetRunningSessions does <c>Getting a all running sessions from the server</c> returns <returns>A Dictionary<string, string> containing all users as key and a value of all data</returns> sends data using SendDataToTCP and then Receive it using ReceiveFromTcp</summary>
        ///
        
```
comment volgens styleguide